plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java

```diff
@@ -202,25 +208,50 @@ public static void waitForProjects(Runnable delayTactic, IProject... projects) {
         Collections.addAll(jobs, jobManager.find("org.eclipse.wst.server.core.family"));
         Collections.addAll(jobs, jobManager.find("org.eclipse.wst.server.ui.family"));
         Collections.addAll(jobs, jobManager.find(ValidationBuilder.FAMILY_VALIDATION_JOB));
+
+        // Remove SLEEPING jobs as they may not run for a long time.
+        // May be specific to FAMILY_AUTO_BUILD jobs.
+        for (Iterator<Job> iter = jobs.iterator(); iter.hasNext();) {
+          Job job = iter.next();
+          if (job.getState() != Job.RUNNING && job.getState() != Job.WAITING) {
+            iter.remove();
+          }
+        }
+
+        // track whether we've reached a fixpoint in the build errors
         allBuildErrors = getAllBuildErrors(projects);
-        if (DEBUG) {
+        buildErrorsChanging =
```

Comment:
Would it make sense to try to check whether the build errors are the same as in the previous round instead of just the change in the count?